### PR TITLE
 Fix Unable to upload avatar through the app

### PR DESCRIPTION
**Pull request by ani4aniket**
@RocketChat/ReactNative

Closes Issue #1370  #1409  #441 

Screen Recording 2019-11-23 at 4 49 44 AM

### CODE REVIEW

File: app/containers/Avatar.js [hunk 1]
```diff
@@ -1,80 +1,103 @@
...
-const Avatar = React.memo(({
-	text, size, baseUrl, borderRadius, style, avatar, type, children, userId, token
-}) => {
-	const avatarStyle = {
-		width: size,
-		height: size,
-		borderRadius
-	};
+const formatUrl = (url, baseUrl, uriSize, avatarAuthURLFragment) =>
+  `${baseUrl}${url}?format=png&width=${uriSize}&height=${uriSize}${avatarAuthURLFragment}&t=${timestamp}`;
```
Comment (diegolmello) [inline app/containers/Avatar.js]: By using a timestamp on the link, you're breaking the entire avatar cache.
Avatar will start to load properly, but they will be requested from network all the time.

Comment (ani4aniket) [reply app/containers/Avatar.js]: I have fixed the lint errors which is caused by Prettier. Please review it.

**Comment by TDHTTTT** [comment]
Thanks a lot for the PR. Any news for merging or anything I can help with?

**Comment by djorkaeffalexandre** [comment]
Hey guys, I'll just close this because of inactivity.
This can't be merged because it's not a good solution, if we use this, the avatars will be requested again at each time that appears at the screen.
We've tried to control the cache better but didn't find a good solution, so, we're waiting for a solution that will be implemented at our server side, with dynamics avatar urls.
Thanks anyway. 😁

**Comment by danfelbm** [comment]
This issue can be traced back to 2018! It's been a long time since this issue was first encountered. I wonder how this hasn't been addressed (I'd have thought people would complain more about not being able to change their avatars)

**Comment by shaheem-khanzada** [comment]
@djorkaeffalexandre @diegolmello I think its not an image cache problem because as I test it and find out that we are using React.memo in avatar component so that why our avatar is not getting re-render because previous and next username is same

**Comment by diegolmello** [comment]
@shaheem-khanzada Can you open a PR fixing it then?
Thanks!

**Pull request closed without merging**